ext/ddtrace.c

```diff
@@ -374,20 +375,40 @@ bool ddtrace_alter_sampling_rules_file_config(zval *old_value, zval *new_value,
     return dd_save_sampling_rules_file_config(Z_STR_P(new_value), PHP_INI_USER, PHP_INI_STAGE_RUNTIME);
 }
 
-static inline void dd_alter_prop(size_t prop_offset, zval *old_value, zval *new_value, zend_string *new_str) {
+static inline void dd_alter_prop_common(size_t prop_offset, zval *old_value, zval *new_value, zend_string *new_str, bool skip_inferred) {
     UNUSED(old_value, new_str);
 
     ddtrace_span_properties *pspan = ddtrace_active_span_props();
     while (pspan) {
+        if (skip_inferred) {
+            ddtrace_span_data *span = SPANDATA(pspan);
+            if (span && span->type == DDTRACE_INFERRED_SPAN) {
```

Comment:
```suggestion
            if (span->type == DDTRACE_INFERRED_SPAN) {
```
Nit, but SPANDATA() is just an offset calculation from pspan and will never be NULL.